src/XMakeTasks/Microsoft.CSharp.CurrentVersion.targets

```diff
@@ -46,6 +46,18 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         <SynthesizeLinkMetadata Condition=" '$(SynthesizeLinkMetadata)' == '' and '$(HasSharedItems)' == 'true' ">true</SynthesizeLinkMetadata>
     </PropertyGroup>
 
+    <PropertyGroup>
+        <!-- Select a default C# compiler here, if unspecified, instead of in the Csc task, so that we can also
+             decide on the debug extension to be used. -->
+        <CscToolExe Condition="'$(CscToolExe)' == '' and '$(MSBuildRuntimeType)' != 'Mono'">csc.exe</CscToolExe>
```

Comment:
Hm I need this here to figure out a debug file extension to use. I could rework this to preserve the original `CscToolExe` in case of !Mono . Would that be acceptable? For Mono, the compiler server is not being used yet.